Tests/FoundationMacrosTests/MacroTestUtilities.swift

```diff
@@ -138,5 +147,32 @@ func AssertMacroExpansion(macros: [String : Macro.Type], testModuleName: String
 }
 
 func AssertPredicateExpansion(_ source: String, _ result: String = "", diagnostics: Set<DiagnosticTest> = [], file: StaticString = #file, line: UInt = #line) {
-    AssertMacroExpansion(macros: ["Predicate": PredicateMacro.self], source, result, diagnostics: diagnostics, file: file, line: line)
+    AssertMacroExpansion(
+        macros: ["Predicate": PredicateMacro.self],
+        source,
+        result,
+        diagnostics: diagnostics,
+        file: file,
+        line: line
+    )
+    AssertMacroExpansion(
+        macros: ["Expression" : FoundationMacros.ExpressionMacro.self],
+        source._replacing("#Predicate", with: "#Expression"),
+        result._replacing(".Predicate", with: ".Expression"),
+        diagnostics: Set(diagnostics.map(\.mappedToExpression)),
+        file: file,
+        line: line
+    )
+}
+
+extension String {
+    func _replacing(_ text: String, with other: String) -> Self {
+        if #available(macOS 13.0, *) {
+            // Use the stdlib API if available
+            self.replacing(text, with: other)
```

Comment:
Nit: Is this the function from [String Processing](https://github.com/apple/swift-experimental-string-processing/blob/cd0ce57a69310b419218f26f1656f206d5995c3c/Sources/_StringProcessing/Algorithms/Algorithms/Replace.swift#L89)? If so, would it make sense to match their availability with `@available(SwiftStdlib 5.7, *)` instead of `macOS 13.0`?

Reply:
Yes, but `SwiftStdlib` is a macro available only within the Swift stdlib projects (just like `FoundationPreview` is only available within our project) so unfortunately we can't use that from here

Reply:
Gotcha, thanks!